src/frontend/planner_test/tests/testdata/output/over_window_function.yaml

```diff
@@ -855,6 +855,110 @@
+  sql: |
+    CREATE TABLE t (a INT, b INT, c INT);
+    SELECT a, b, c
+    FROM (
+      SELECT
+        a, b, c,
+        ROW_NUMBER() OVER (PARTITION BY a ORDER BY b) - 1 AS rn
+      FROM t
+    )
+    WHERE rn < 10;
+  logical_plan: |-
+    LogicalProject { exprs: [t.a, t.b, t.c] }
+    └─LogicalFilter { predicate: ($expr1 < 10:Int32) }
+      └─LogicalProject { exprs: [t.a, t.b, t.c, (row_number - 1:Int32) as $expr1] }
+        └─LogicalOverWindow { window_functions: [row_number() OVER(PARTITION BY t.a ORDER BY t.b ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)] }
+          └─LogicalProject { exprs: [t.a, t.b, t.c, t._row_id, t._rw_timestamp] }
+            └─LogicalScan { table: t, columns: [t.a, t.b, t.c, t._row_id, t._rw_timestamp] }
+  optimized_logical_plan_for_batch: |-
+    LogicalTopN { order: [t.b ASC], limit: 10, offset: 0, group_key: [t.a] }
+    └─LogicalScan { table: t, columns: [t.a, t.b, t.c] }
+  optimized_logical_plan_for_stream: |-
+    LogicalTopN { order: [t.b ASC], limit: 10, offset: 0, group_key: [t.a] }
+    └─LogicalScan { table: t, columns: [t.a, t.b, t.c] }
+- name: test arithmetic with addition
+  sql: |
+    CREATE TABLE t (a INT, b INT, c INT);
+    SELECT a, b, c
+    FROM (
+      SELECT
+        a, b, c,
+        ROW_NUMBER() OVER (PARTITION BY a ORDER BY b) + 5 AS rn_plus_five
+      FROM t
+    )
+    WHERE rn_plus_five = 6;
+  logical_plan: |-
+    LogicalProject { exprs: [t.a, t.b, t.c] }
+    └─LogicalFilter { predicate: ($expr1 = 6:Int32) }
+      └─LogicalProject { exprs: [t.a, t.b, t.c, (row_number + 5:Int32) as $expr1] }
+        └─LogicalOverWindow { window_functions: [row_number() OVER(PARTITION BY t.a ORDER BY t.b ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)] }
+          └─LogicalProject { exprs: [t.a, t.b, t.c, t._row_id, t._rw_timestamp] }
+            └─LogicalScan { table: t, columns: [t.a, t.b, t.c, t._row_id, t._rw_timestamp] }
+  optimized_logical_plan_for_batch: |-
+    LogicalTopN { order: [t.b ASC], limit: 1, offset: 0, group_key: [t.a] }
+    └─LogicalScan { table: t, columns: [t.a, t.b, t.c] }
+  optimized_logical_plan_for_stream: |-
+    LogicalTopN { order: [t.b ASC], limit: 1, offset: 0, group_key: [t.a] }
+    └─LogicalScan { table: t, columns: [t.a, t.b, t.c] }
+- name: test optimization when rn column is kept
+  sql: |
+    CREATE TABLE t (a INT, b INT, c INT);
+    SELECT a, b, c, rn_plus_one
+    FROM (
+      SELECT
+        a, b, c,
+        ROW_NUMBER() OVER (PARTITION BY a ORDER BY b) + 1 AS rn_plus_one
+      FROM t
+    )
+    WHERE rn_plus_one = 2;
+  logical_plan: |-
+    LogicalProject { exprs: [t.a, t.b, t.c, $expr1] }
+    └─LogicalFilter { predicate: ($expr1 = 2:Int32) }
+      └─LogicalProject { exprs: [t.a, t.b, t.c, (row_number + 1:Int32) as $expr1] }
+        └─LogicalOverWindow { window_functions: [row_number() OVER(PARTITION BY t.a ORDER BY t.b ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)] }
```

Comment:
Potential for future optimization? Seems like `rn_plus_one` is always `2`. I guess this can just be a project